sympy/functions/special/delta_functions.py

```diff
@@ -194,7 +197,18 @@ def eval(cls, arg, k=0):
         if arg.is_nonzero:
             return S.Zero
         if fuzzy_not(im(arg).is_zero):
-            raise ValueError("Function defined only for Real Values. Complex part: %s  found in %s ." % (repr(im(arg)), repr(arg)) )
+            raise ValueError(filldedent('''
+                Function defined only for Real Values.
+                Complex part: %s  found in %s .''' % (
+                repr(im(arg)), repr(arg))))
+        c, nc = arg.args_cnc()
+        if c and c[0] == -1:
+            # keep this fast and simple instead of using
+            # could_extract_minus_sign
+            if k % 2 == 1:
+                return -cls(-arg, k)
+            elif k % 2 == 0:
+                return cls(-arg)
```

Comment:
I believe `k` should be here as well.

Reply:
Thanks! I added a test and updated the function.